### PR TITLE
Remove pypy from tox.ini

### PR DESCRIPTION
**Pull request by jdufresne**
The pypy executable is Python 2 and Python 2 support was dropped in commit 9412bd76b7f5fdf0adc231ef8130e45bda130164.

### CODE REVIEW

**Comment by tk0miya** [comment]
Does this mean dropping all of pypy support?
The commit only meant we'll drop python2.x support in Sphinx-2.0. So we'll still keep supporting pypy-3.x. I don't know much about pypy. So I'm afraid of this breaks it.

**Comment by jdufresne** [comment]
I first noticed this as an issue when using `tox` to run _all_ test environments. Please observe:

```
$ tox -e pypy
pypy create: /home/jon/devel/sphinx/.tox/pypy
pypy installdeps: .[test,websupport]
ERROR: invocation failed (exit code 1), logfile: /home/jon/devel/sphinx/.tox/pypy/log/pypy-1.log
ERROR: actionid: pypy
msg: getenv
cmdargs: ['/home/jon/devel/sphinx/.tox/pypy/bin/pip', 'install', '.[test,websupport]']

Processing /home/jon/devel/sphinx
    Complete output from command python setup.py egg_info:
    ERROR: Sphinx requires at least Python 3.5 to run.
```

This is because the environment `pypy` represents Python 2.7.

```
$ pypy --version
Python 2.7.13 (0e7ea4fe15e82d5124e805e2e4a37cae1a402d4b, Apr 12 2018, 14:50:12)
[PyPy 5.10.0 with GCC 8.0.1 20180324 (Red Hat 8.0.1-0.20)]
```

If you'd like to include a Python 3 compatible PyPy, that would be pypy3 or pypy3.5:

```
$ pypy3 --version
Python 3.5.3 (89428233efed, Apr 12 2018, 16:18:00)
[PyPy 5.10.1 with GCC 8.0.1 20180324 (Red Hat 8.0.1-0.20)]
```

However, as noted in a [#5342 comment](https://github.com/sphinx-doc/sphinx/pull/5342#discussion_r213893751), this also fails, but for different reasons:

```
$ tox -e pypy3

...

Installing collected packages: six, MarkupSafe, Jinja2, Pygments, docutils, snowballstemmer, pytz, babel, alabaster, imagesize, chardet, idna, urllib3, certifi, requests, pyparsing, packaging, pbr, mock, py, attrs, more-itertools, atomicwrites, pluggy, pathlib2, pytest, coverage, pytest-cov, webencodings, html5lib, pyflakes, pycodestyle, mccabe, flake8, flake8-import-order, typed-ast, mypy, sqlalchemy, whoosh, Sphinx
  Running setup.py install for typed-ast: started
    Running setup.py install for typed-ast: finished with status 'error'
    Complete output from command /home/jon/devel/sphinx/.tox/pypy3/bin/pypy3 -u -c "import setuptools, tokenize;__file__='/tmp/pip-install-wb7k5l7b/typed-ast/setup.py';f=getattr(tokenize, 'open', open)(__file__);code=f.read().replace('\r\n', '\n');f.close();exec(compile(code, __file__, 'exec'))" install --record /tmp/pip-record-_l6e9xc9/install-record.txt --single-version-externally-managed --compile --install-headers /home/jon/devel/sphinx/.tox/pypy3/include/site/python3.5/typed-ast:
    /home/jon/devel/sphinx/.tox/pypy3/lib-python/3/distutils/__init__.py:4: PendingDeprecationWarning: the imp module is deprecated in favour of importlib; see the module's documentation for alternative uses
      import imp
    running install
    running build
    running build_py
    creating build
    creating build/lib.linux-x86_64-3.5
    creating build/lib.linux-x86_64-3.5/typed_ast
    copying typed_ast/conversions.py -> build/lib.linux-x86_64-3.5/typed_ast
    copying typed_ast/ast3.py -> build/lib.linux-x86_64-3.5/typed_ast
    copying typed_ast/ast27.py -> build/lib.linux-x86_64-3.5/typed_ast
    copying typed_ast/__init__.py -> build/lib.linux-x86_64-3.5/typed_ast
    running build_ext
    building '_ast27' extension
    creating build/temp.linux-x86_64-3.5
    creating build/temp.linux-x86_64-3.5/ast27
    creating build/temp.linux-x86_64-3.5/ast27/Parser
    creating build/temp.linux-x86_64-3.5/ast27/Python
    creating build/temp.linux-x86_64-3.5/ast27/Custom
    cc -pthread -DNDEBUG -O2 -fPIC -Iast27/Include -I/usr/lib64/pypy3-5.10/include -c ast27/Parser/acceler.c -o build/temp.linux-x86_64-3.5/ast27/Parser/acceler.o
    ast27/Parser/acceler.c:13:10: fatal error: pgenheaders.h: No such file or directory
     #include "pgenheaders.h"
              ^~~~~~~~~~~~~~~
    compilation terminated.
    error: command 'cc' failed with exit status 1
```

This looks like pypy3 is missing some C headers expected by typed-ast. But I haven't dived deeper.

Given this, it looks like Sphinx doesn't actually support any version of PyPy. It doesn't support pypy because it is Python 2 and it doesn't support pypy3 due to the C header issue. I understand that isn't the intention of the project, but it seems to be the logical result. This is fine with me, I don't use pypy, I'm just trying to run the Sphinx tests.

> Does this mean dropping all of pypy support?

This change does not affect the status of pypy3. That is up to you and any other maintainers. I'm not suggesting the project drop pypy3, just responding to the fact that Python2 was intentionally dropped as an environment. How pypy3 is handled is a different matter.

In summary, I think pypy should be removed from tox and pypy3 should be added once it is working.

**Comment by tk0miya** [comment]
I just understood `py` target is only used as pypy2. Okay, let's remove it!
